test/unit/test_SolverBranchBoundary.cc

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
We should maybe have a separate `parallel-test/unit/` folder so it's easier and more modular to enable / disable depending on whether we're compiling the parallel code or not.

Reply:
Sure, good idea!